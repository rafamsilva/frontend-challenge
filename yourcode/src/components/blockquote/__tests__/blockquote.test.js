import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

import Blockquote from '../';

describe('Blockquote Component', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  describe('Render', () => {
    it('should render without crashing', () => {
      act(() => {
        render(<Blockquote title="Plot" />, container);
      });
    });

    it('should render with text', () => {
      act(() => {
        render(<Blockquote title="Plot" text="Hello World!" />, container);
      });

      expect(container.querySelector("[data-testid='title']").textContent).toBeDefined();
      expect(container.querySelector("[data-testid='title']").textContent).toBe('Plot');

      expect(container.querySelector("[data-testid='text']").textContent).toBeDefined();
      expect(container.querySelector("[data-testid='text']").textContent).toBe('Hello World!');

      expect(container.querySelector("[data-testid='list']")).toBeNull();
    });

    it('should render with list', () => {
      act(() => {
        render(<Blockquote title="Directors" list={['Darwin', 'Rafael']} />, container);
      });

      expect(container.querySelector("[data-testid='title']").textContent).toBeDefined();
      expect(container.querySelector("[data-testid='title']").textContent).toBe('Directors');

      expect(container.querySelector("[data-testid='list']")).toBeDefined();
      expect(container.querySelector("[data-testid='list']").children).toHaveLength(2);

      expect(container.querySelector("[data-testid='text']")).toBeNull();
    });
  });

  describe('Snapshot', () => {
    it('should match snapshot with text', () => {
      act(() => {
        render(<Blockquote title="Plot" text="Hello World!" />, container);
      });

      expect(pretty(container.innerHTML)).toMatchSnapshot();
    });

    it('should match snapshot with list', () => {
      act(() => {
        render(<Blockquote title="Directors" list={['Darwin', 'Rafael']} />, container);
      });

      expect(pretty(container.innerHTML)).toMatchSnapshot();
    });
  });
});
