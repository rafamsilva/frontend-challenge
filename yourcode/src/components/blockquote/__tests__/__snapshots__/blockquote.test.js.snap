// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote Component Snapshot should match snapshot with list 1`] = `
"<div class=\\"sc-AxjAm StDqN\\">
  <h4 data-testid=\\"title\\" class=\\"sc-AxirZ ertLcl\\">Directors</h4>
  <ul data-testid=\\"list\\" class=\\"sc-AxhCb ccMvAG\\">
    <li class=\\"sc-AxiKw sc-AxhUy bDXEwN\\">Darwin</li>
    <li class=\\"sc-AxiKw sc-AxhUy bDXEwN\\">Rafael</li>
  </ul>
</div>"
`;

exports[`Blockquote Component Snapshot should match snapshot with text 1`] = `
"<div class=\\"sc-AxjAm StDqN\\">
  <h4 data-testid=\\"title\\" class=\\"sc-AxirZ ertLcl\\">Plot</h4>
  <p data-testid=\\"text\\" class=\\"sc-AxiKw dbORnj\\">Hello World!</p>
</div>"
`;
